File: Assembly-FSharp/Assembly-FSharp/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/MovementModel.fs
namespace Styx.Models

open UnityEngine

[<CreateAssetMenu(fileName = "Movement Model", menuName = "Models/Movement model")>]
type MovementModel() =
    inherit ScriptableObject()

    [<SerializeField>]
    let mutable linearSpeed: float32 = 15.f
    [<SerializeField>]
    let mutable rotationSpeed: float32 = 500.f

    member this.LinearSpeed 
        with get() = linearSpeed
        and set(v) = linearSpeed <- v
    member this.RotationSpeed
        with get() = rotationSpeed
        and set(v) = rotationSpeed <- v

File: Assets/_Scripts/Models/MovementModel.cs
using UnityEngine;

namespace Styx.Models
{
    [CreateAssetMenu(fileName = "Movement Model", menuName = "Models/Movement model")]
    public class MovementModel : ScriptableObject
    {
        [SerializeField]
        private float _linearSpeed = 15f;
        [SerializeField]
        private float _rotationSpeed = 500f;

        public float LinearSpeed
        {
            get { return _linearSpeed; }
            set { _linearSpeed = value; }
        }

        public float RotationSpeed
        {
            get { return _rotationSpeed; }
            set { _rotationSpeed = value; }
        }
    }
}
